src/components/WorkOrder/EditInventoryItemsFields.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from "react";
+import Select from "react-select";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faSpinner } from "@fortawesome/free-solid-svg-icons";
+
+import { FIELDS } from "./formConfig";
+import api from "../../queries/api";
+import {
+  getWorkTypeScheduledWorkOptions,
+  convertKnackResponseObjectToSelectFormOption,
+  convertKnackResponseStringToSelectFormOption,
+} from "../../queries/knackObjectHelpers";
+
+export default class EditInventoryItemsFields extends Component {
+  _isMounted = false;
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      itemOptions: [],
+      existingFormData: null,
+    };
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.getItemOptions();
+    this.getInventoryItemFields();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  getInventoryItemFields = () => {
+    api
+      .workOrder()
+      .getEditInventory(this.props.inventoryItemId)
+      .then(res => {
+        const inventoryItemData = res.data.records[0];
+        this._isMounted &&
+          this.setState({ existingFormData: inventoryItemData });
+      });
+  };
+
+  getItemOptions = () => {
+    api
+      .workOrder()
+      .getInventoryItems()
+      .then(res => {
+        const itemOptions = res.data.records.map(item => {
+          return { label: item.identifier, value: item.id };
+        });
+        this._isMounted && this.setState({ itemOptions });
+      });
+  };
+
+  handleConditionChange = e => {
+    let data = {};
+    data[FIELDS.WORK_ORDER_ITEM_CONDITION] = e.target.value;
+    this.props.handleItemPropertyChange(data);
+  };
+
+  handleQuantityChange = e => {
+    let data = {};
+    data[FIELDS.WORK_ORDER_ITEM_QUANTITY] = e.target.value;
+    this.props.handleItemPropertyChange(data);
+  };
+
+  handleSourceChange = option => {
+    let data = {};
+    data[FIELDS.WORK_ORDER_ITEM_SOURCE] = option.value;
+    this.props.handleItemPropertyChange(data);
+  };
+
+  handleCommentChange = e => {
```

Comment:
I'm seeing a lot in common between the `handleConditionChange`, `handleQuantityChange`, & `handleCommentChange` functions. What would you think about adding a parameter for the field and merging these into one function?

Reply:
Glad that you brought this up! After looking again, `handleConditionChange` isn't even called anymore so I removed it and combined the other two as `handleFieldChange` with a `field` parameter.